### ISSUE TITLE
[java] GuardLogStatement: Should not need to guard parameterized log messages where the replacement arg is "this"

### ISSUE THREAD

**Issue by jbisotti** [pmd/pmd#5152]
**Affects PMD Version:** 
7.0.0

**Rule:**
[GuardLogStatement](https://pmd.github.io/pmd/pmd_rules_java_bestpractices.html#guardlogstatement)

**Description:**
I do not believe an argument, which is `this`, should result in a violation of this rule.

**Code Sample demonstrating the issue:**

```java
        LOG.info(
                "Some message here : foo={}",
                this
        );
```

**Expected outcome:**

PMD reports "... Rule:GuardLogStatement Priority:2 Logger calls should be surrounded by log level guards..".
I think that's a false positive.

**Running PMD through:** *[CLI | Ant | Maven | Gradle | Designer | Other]*
Run via maven-pmd-plugin (3.24.0)
